SingularityBase/src/main/java/com/hubspot/mesos/SingularityMesosInfo.java

```diff
@@ -0,0 +1,59 @@
+package com.hubspot.mesos;
+
+import java.util.Objects;
+
+import org.apache.mesos.Protos.Image;
+import org.apache.mesos.Protos.Image.Type;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.wordnik.swagger.annotations.ApiModelProperty;
+
+public class SingularityMesosInfo {
+  private final String image;
+  private final Image.Type type;
+
+  @JsonCreator
+  public SingularityMesosInfo(@JsonProperty("image") String image, // TODO should this be more like the Mesos image protobuf?
```

Comment:
I haven't taken too close of a look at the mesos containerizer stuff for docker. Does it have similar volume mount/port mapping type options to the docker containerizer ones? Would be good to have a way to support those as well

Reply:
Added in the updated PR